src/Workspaces/Core/Portable/Diagnostics/DiagnosticAnalysisResultBuilder.cs

```diff
@@ -113,7 +113,12 @@ public void AddExternalSemanticDiagnostics(DocumentId documentId, IEnumerable<Di
 
         private void AppendDiagnostics(ref Dictionary<DocumentId, List<DiagnosticData>> map, Document documentOpt, Diagnostic diagnostic)
         {
-            if (documentOpt?.SupportsDiagnostics() == false)
+            if (documentOpt is null)
+            {
+                return;
+            }
+
+            if (!documentOpt.SupportsDiagnostics())
```

Comment:
You could write
``` c#
            if (documentOpt?.SupportsDiagnostics() != true)
            {
                return;
            }
```

Reply:
I could, but I figured if I already messed it up once it's not obvious enough to prefer.